ctlcmd/cmdbase.py

```diff
@@ -771,7 +773,189 @@ def globcomp(text):
+  Placeholders will be specified in the angle braced. For the full list of
+  special placeholders, as well as how the file parsing is handled, see the
+  detailed documentation in the `ctlcmd.cmdbase.savefilecmd.parse` method.
+  """
+  DEFAULT_ROOTFILE = 'SAVEROOT_<TIMESTAMP>'
+  def __init__(self, cmd):
+    controlcmd.__init__(self, cmd)
+     
+
+  def add_args(self):
+    group = self.parser.add_argument_group(
+        "root file saving options", """Options for changing the root file location.
+        For more details, see the official documentation.""")
+    group.add_argument('--saveroot',
+                       type=str,
+                       default=self.DEFAULT_ROOTFILE,
+                       help="""File path to save (placeholders in angle braces).
+                       default=%(default)s""")
+ 
+  def parse(self,args):
+    if not args.saveroot:  # Early exit if savefile is not set
+      self.saveroot = None
+      return args
+    filename = args.saveroot
+
+    # Adding time stamp filenames
+    timestring = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+    filename = re.sub('<TIMESTAMP>', timestring, filename, flags=re.IGNORECASE)
+
+    # Adding boardid to the settings
+    filename = re.sub('<BOARDID>',
+                      str(self.board.boardid),
+                      filename,
+                      flags=re.IGNORECASE)
+    # Adding boardtype to the settings
+    filename = re.sub('<BOARDTYPE>',
+                      str(self.board.boardtype),
+                      filename,
+                      flags=re.IGNORECASE)
 
+    # Substituting tokens for argument values
+    for action in self.parser._actions:
+      for string in action.option_strings:
+        string = string.strip('-')
+        if not hasattr(args, string): continue
+        val = ''
+        try:
+          val = action.type(getattr(args, string))
+        except:
+          pass
+        substring = '{0}{1}'.format(string, val)
+        substring = re.sub(r'\.', 'p', substring)
+        filename = re.sub('\\<{0}\\>'.format(string),
+                          substring,
+                          filename,
+                          flags=re.IGNORECASE)
+    self.saveroot = filename
+    return args
+      
+  def maketree(self,data,datatypes={}):
+    """
+    @brief Create the root file and the TTree in the root file
+
+    @details This command creates the root file and creates the TTree with the correct
+    data types and names for each branch of the tree. This also saves some extra basic data 
+    like the time the file was created, the Board_ID and Board_Type. This function is only called once.
+
+    """
+     
+    self.rootfile = uproot.recreate(self.saveroot)
+    standard_dict={"time":np.float32,"det_id":int,"gantry x":np.float32,"gantry y":np.float32,
+                   "gantry z":np.float32,"LED bias voltage":np.float32,"LED temp":np.float32,"SiPM temp":np.float32} 
```

Comment:
For the sake of users potentially accessing the data information within the control program (ex: we might want to have the users check a previous calibration result for on-the-fly plotting or coordinates extraction), it would be nicer to have the branches names be compatible with a variable name, so users can access information like `data_tree.det_id` Let us stick to the Google python style guide [1]. For the time being. So:

- `gantry x` -> `gantry_x`
- `LED_bias_voltage` -> `led_bv`
- `LED temp` -> `led_temp`
- `SiPM temp` -> `sipm_temp`  






[1] https://google.github.io/styleguide/pyguide.html#316-naming